src/components/two-factor-authentication/SelfServiceWorkflow.vue

```diff
@@ -0,0 +1,263 @@
+<script setup lang="ts">
+  import { computed, ref, type ComputedRef, type Ref } from 'vue';
+  import { useDisplay } from 'vuetify';
+  import LayoutCard from '@/components/cards/LayoutCard.vue';
+  import SpshTooltip from '@/components/admin/SpshTooltip.vue';
+  import { useI18n, type Composer } from 'vue-i18n';
+  import {
+    useTwoFactorAuthentificationStore,
+    type TwoFactorAuthentificationStore,
+  } from '@/stores/TwoFactorAuthentificationStore';
+
+  const { mdAndDown }: { mdAndDown: Ref<boolean> } = useDisplay();
+  const { t }: Composer = useI18n({ useScope: 'global' });
+  const twoFactorStore: TwoFactorAuthentificationStore = useTwoFactorAuthentificationStore();
+
+  enum TwoFactorSteps {
+    Start = 'start',
+    QRCode = 'qrcode',
+    Verify = 'verify',
+  }
+
+  const workflowStep: Ref<TwoFactorSteps> = ref(TwoFactorSteps.Start);
+  const errorMessage: Ref<string> = ref('');
+  const otp: Ref<string> = ref('');
+
+  const dialogHeader: ComputedRef<string> = computed(() => {
+    switch (workflowStep.value) {
+      case TwoFactorSteps.Start:
+        return t('admin.person.twoFactorAuthentication.setUpLong');
+      case TwoFactorSteps.QRCode:
+        return t('admin.person.twoFactorAuthentication.softwareTokenOption');
+      case TwoFactorSteps.Verify:
+        return t('admin.person.twoFactorAuthentication.header');
+      default:
+        return t('admin.person.twoFactorAuthentication.setUpLong');
+    }
+  });
+
+  type Emits = {
+    (event: 'dialogClosed'): void;
+  };
+
+  const emits: Emits = defineEmits<{
+    (event: 'dialogClosed'): void;
+  }>();
+
+  type Props = {
+    personId: string;
+  };
+
+  const props: Props = defineProps<Props>();
+
+  async function close2FADialog(isActive: Ref<boolean>): Promise<void> {
+    if (workflowStep.value !== TwoFactorSteps.Start) {
+      emits('dialogClosed');
+      twoFactorStore.qrCode = '';
+      twoFactorStore.hasToken = null;
+      twoFactorStore.tokenKind = null;
+    }
+
+    isActive.value = false;
+    workflowStep.value = TwoFactorSteps.Start;
+  }
+
+  async function proceed(isActive: Ref<boolean>): Promise<void> {
```

Comment:
```suggestion
  async function proceedToNextWorkflowStep(isActive: Ref<boolean>): Promise<void> {
```

Zur Schönheit aber auch klarer